Runner/Sandbox.cs

```diff
@@ -69,25 +69,25 @@ public ExecutionResult ExecuteMethod(MethodInfo method, params object[] args)
         public IHookRegistry GetHookRegistry()
         {
             var hookRegistry = new HookRegistry(this);
-            hookRegistry.AddBeforeSuiteHooks(GetAllMethodsForSpecAssemblies(typeof(BeforeSuite).ToString()));
-            hookRegistry.AddAfterSuiteHooks(GetAllMethodsForSpecAssemblies(typeof(AfterSuite).ToString()));
-            hookRegistry.AddBeforeSpecHooks(GetAllMethodsForSpecAssemblies(typeof(BeforeSpec).ToString()));
-            hookRegistry.AddAfterSpecHooks(GetAllMethodsForSpecAssemblies(typeof(AfterSpec).ToString()));
-            hookRegistry.AddBeforeScenarioHooks(GetAllMethodsForSpecAssemblies(typeof(BeforeScenario).ToString()));
-            hookRegistry.AddAfterScenarioHooks(GetAllMethodsForSpecAssemblies(typeof(AfterScenario).ToString()));
-            hookRegistry.AddBeforeStepHooks(GetAllMethodsForSpecAssemblies(typeof(BeforeStep).ToString()));
-            hookRegistry.AddAfterStepHooks(GetAllMethodsForSpecAssemblies(typeof(AfterStep).ToString()));
+            hookRegistry.AddBeforeSuiteHooks(AssemblyScanner.AttributeToMethodInfo[typeof(BeforeSuite)]);
```

Comment:
Could make the `AttributeToMethodInfo` a private field and have a `GetHookMethod` method that will return a list/collection of `MethodInfo`. This method could either take a `Type` param or be a Generic method.